ibc/handle_message_test.go

```diff
@@ -0,0 +1,226 @@
+package ibc
+
+import (
+	"testing"
+
+	ibcTypes "github.com/pokt-network/pocket/ibc/types"
+	"github.com/pokt-network/pocket/persistence/indexer"
+	"github.com/pokt-network/pocket/shared/codec"
+	coreTypes "github.com/pokt-network/pocket/shared/core/types"
+	"github.com/pokt-network/pocket/shared/crypto"
+	"github.com/stretchr/testify/require"
+)
+
+func TestHandleMessage_ErrorAlreadyInMempool(t *testing.T) {
+	// Prepare test data
+	_, tx := prepareUpdateMessage(t, []byte("key"), []byte("value"))
+	txProtoBytes, err := codec.GetCodec().Marshal(tx)
+	require.NoError(t, err)
+
+	// Prepare the environment
+	_, utilityMod, _, _ := prepareEnvironment(t, 1, 0, 0, 0)
+
+	// Manually add the tx to the mempool
+	err = utilityMod.GetMempool().AddTx(txProtoBytes)
+	require.NoError(t, err)
+
+	// Error on having a duplciate transaction
+	err = utilityMod.HandleTransaction(txProtoBytes)
+	require.Error(t, err)
+	require.EqualError(t, err, coreTypes.ErrDuplicateTransaction().Error())
+}
+
+func TestHandleMessage_ErrorAlreadyCommitted(t *testing.T) {
+	// Prepare the environment
+	_, utilityMod, persistenceMod, _ := prepareEnvironment(t, 0, 0, 0, 0)
+	idxTx := prepareIndexedMessage(t, persistenceMod.GetTxIndexer())
+
+	// Error on having an indexed transaction
+	err := utilityMod.HandleTransaction(idxTx.Tx)
+	require.Error(t, err)
+	require.EqualError(t, err, coreTypes.ErrTransactionAlreadyCommitted().Error())
+}
+
+func TestHandleMessage_BasicValidation_Message(t *testing.T) {
+	updateMsg, _ := prepareUpdateMessage(t, []byte("key"), []byte("value"))
```

Comment:
For this tet and the one below, I don't fully understand why you have a "mini test" above the `testCases`. Can you add some comments?